### ISSUE TITLE
Network gateway not sending incoming UDP datagrams to app

### ISSUE THREAD

**Issue by erikvk** [Starcounter/Home#438]
_This issue is for reference – has already been discussed with @alemoi who is investigating_

When having an `Handle.Udp()` in an app, listening for UDP datagrams in a production environment, we have noticed that the Starcounter network gateway can sometimes stop sending the incoming datagrams to the Starcounter app. When this happens, a network monitor tool like Wireshark or TCPView will still register the incoming datagrams, and even associate them with the `scnetworkgateway` process, but the UDP handle in the app code is not called.

Restarting the codehost (including restarting `sccode`) has no effect. Running `staradmin kill all`, and restarting the Starcounter service and application, however, fixes the issue. 

Our conclusion is that the issue is located in the network gateway.

**Comment by alemoi** [comment]
Just a status update: adding diagnostics and actively trying to figure out the problem.

**Comment by dan31** [comment]
@erikvk @alemoi is it right you could not reproduce it anymore?

**Comment by erikvk** [comment]
@dan31 That's right. I am no longer able to reproduce the issue on our server. I have reached out to our customer, to see if they still have the issue.

My first guess is that this is somehow related to our code, in a way I'm unable to find, and that some update during the past weeks fixed it. Seems like a long shot, but still.

My second guess is that this is somehow time-related. We saw all hangs during the early hours of the day (2:00-6:00 AM), and we had the issue in august/early september, but not now. Also seems like a long shot.

I will let yoy know if this super elusive issue presents itself again.

Until then: ¯\_(ツ)_/¯
